realm/realm-library/src/main/java/io/realm/internal/FieldDescriptor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal;
+
+import io.realm.RealmFieldType;
+
+public class FieldDescriptor {
+
+    private long[] columnIndices;
+    private RealmFieldType lastFieldType;
+    private String lastFieldName;
+
+    public FieldDescriptor(Table table, String fieldDescription, boolean allowList) {
+        if (fieldDescription == null || fieldDescription.isEmpty()) {
+            throw new IllegalArgumentException("Non-empty field name must be provided");
+        }
+        if (fieldDescription.startsWith(".") || fieldDescription.endsWith(".")) {
+            throw new IllegalArgumentException("Illegal field name. It cannot start or end with a '.': " + fieldDescription);
+        }
+        if (fieldDescription.contains(".")) {
```

Comment:
I believe that this code is used elsewhere (or potentially can be used elsewhere), and we should promote it to a utility function.